### DISCUSSION TITLE
Drop Roslynpad

### DISCUSSION THREAD

**Discussion by IXLLEGACYIXL** [stride3d/stride#2153] [Contributors]
Reasons for Removal:

Workarounds: 
currently there is a PR to update Roslynpad with workarounds , it should speed up the process of developement but currently its just a pain to keep it up to date

Performance:
the roslynhost is always running in the background for roslynpad, which makes the stride editor heavier than it has to be, for an editor that maybe 1% of the devs use.

Incompatibility with ProjectWatcher: 
The current integration of Rolsynpad with ProjectWatcher has been problematic, resulting in reduced functionality and reliability. The synergy between an IDE and project monitoring tools is crucial for effective project management. Roslynpad is a single project based host, but we need to cover the entire solution for dependencies to upgrade the serializers. I was not able to find a solution to resolve that. The Repair of projectwatcher fixes it for the IDE as the projectwatcher covers a solution, roslynhost covers just projects.
This wasnt important before as the projectwatcher anyway only covered the root project but now its a problem

Functionality:
compared to any other IDE, roslynpad is just insufficient in features that stride implements and its cumbersome to implement pretty much anything in it.

Comparison with Other IDEs: 
Several alternative IDEs offer more powerful features, stability. 
Giving a beginner roslynpad is more depressing than to help him to setup anotehr IDE "oh wow not even the integrated editor works.. what a stupid engine"

**Comment by Kryptos-FR** [comment]
I would be in favour of the removal under the condition of adding an explicit integration with VS Code as a replacement.

* At the very least, an integration in the Game Studio the same way we can open Visual Studio
* And then a compatible extension later for working with shaders

**Reply by IXLLEGACYIXL** [reply]
well there is youness extension

**Reply by Doprez** [reply]
https://github.com/ykafia/sdsl-syntax-highlighter for reference

**Reply by Kryptos-FR** [reply]
It's just a highlighter though, isn't it? It doesn't generate code.

**Reply by Doprez** [reply]
Yes you are correct but I assume it would be the starting point for full VS Code integration.

**Comment by tebjan** [comment]
What are the problems of updating RoslynPad? Which steps are difficult?
Can we consider enable/disable for performance, instead of completely removing it? It works well and is a nice way for beginners to jump in...

**Reply by IXLLEGACYIXL** [reply]
https://github.com/stride3d/stride/pull/2124
thats the upgrade problem 
the second is thtat projectwatcher doesnt work as roslynpad only covers a project and not the entire solution

**Comment by damian-666** [comment]
What about avaloniaedit, the docking.  Dirksters or a avalonia inside, ,
csscript , syntaxer , the soft debugger, in csscript, tabbed dialogs...the
bots can even code   folder watchers thumbnails,   step,in on out inspect,
etc..i use a debbee attach pipe to script and rather just have a soft
debugger that doesn't switch context to an external UI..  unless vscode
has  light weiggt, and can be forked.  Needs competition , plugin
implantation.
